### DISCUSSION TITLE
how do I dynamically create field names to then retrieve their values from $nu.env?

### DISCUSSION THREAD

**Discussion by ghost** [nushell/nushell#4209] [Q&A]
I am building a configuration generator that will extract environment variables and merge them with on disk configuration.
I found how how to do this by invoking `nu` again:
```nu
let $field = (echo "dynamic-field-key" | str screaming-snake-case);
let $value = (nu -c (build-string "echo $nu.env." $field))
echo $"I found:  ($value)"
```
When I run:
```
DYNAMIC_FIELD_KEY="field value" nu /tmp/hi.nu
```
it yields my expected output of:
```
I found:  field value⏎  
```

Do you know of a simpler or alternative way to do this? This feels like self-reflecting shell programming.

**Comment by kubouch** [comment]
I'm not sure I 100% understand the question but you can use `load-env` to dynamically create environment variable names.

**Reply by ghost** [reply]
It is muddled. Thanks for the `load-env` tip that definitely addresses part of my question.
I am trying to do 2 things:
1. create dynamically named variables in a nu script that derive their value from the environment. `load-env` seems perfect for that.
1. merge values coming from a configuration file and the environment. This is the less clear part of my question 😅
